test/integration/features/profile.feature

```diff
@@ -2,104 +2,161 @@
-    - minishift	Running		(Active)
-    """
+  Scenario: As user, I can switch between existing profiles
+    When executing "minishift profile set Test-123" succeeds
+     Then profile "Test-123" should be the active profile
+     And executing "minishift profile set minishift" succeeds
+     Then profile "minishift" should be the active profile
 
-Scenario: User should be able to delete profile 'foo'
-    When executing "minishift profile delete foo --force" succeeds
-    Then stdout should contain
-    """
-    Profile 'foo' deleted successfully
-    """
+  Scenario: As user, I can start Minishift with default profile 'minishift'
+    Given profile "minishift" has state "Does Not Exist"
+    And profile "minishift" is the active profile
+     When executing "minishift start" succeeds
+     Then profile "minishift" should have state "Running"
+     When executing "minishift profile list" succeeds
+     Then stdout should match "-\s*minishift\s*Running\s*\(Active\)"
 
-Scenario: User can not delete default profile 'minishift'
-    When executing "minishift profile delete minishift --force"
-    Then exitcode should equal "1"
-     And stderr should contain
+  Scenario: As user, I can apply independent settings in profile 'foo'
+    When executing "minishift profile set foo" succeeds
+     Then profile "foo" should be the active profile
+     When executing "minishift --profile minishift config get memory" succeeds
+     Then stdout should match "<nil>"
+     And executing "minishift --profile minishift addons list" succeeds
+     And stdout should match "registry-route\s*: disabled\s*P\(0\)"
+     And executing "minishift addons enable registry-route" succeeds
+     And executing "minishift config set memory 5120" succeeds
+     And executing "minishift config get memory" succeeds
+     And stdout should match "5120"
+     When executing "minishift start" succeeds
+     Then stdout should contain
      """
-     Default profile 'minishift' can not be deleted
+     Add-on 'registry-route' created docker-registry route.
      """
+     And profile "foo" should have state "Running"
+     When executing "minishift --profile minishift config get memory" succeeds
+     Then stdout should match "<nil>"
+     And executing "minishift --profile minishift addons list" succeeds
+     And stdout should match "registry-route\s*: disabled\s*P\(0\)"
+     And executing "minishift addons list" succeeds
+     And stdout should match "registry-route\s*: enabled\s*P\(0\)"
+     And executing "minishift ssh -- less /proc/meminfo" succeeds
+     And stdout should match "MemTotal:\s*5[0-1][0-9]{5}\s*kB"
 
-Scenario: Deleting Minishift
-   Given Minishift has state "Running"
-    When executing "minishift delete --force" succeeds
-    Then Minishift should have state "Does Not Exist"
-    When executing "minishift ip"
-    Then exitcode should equal "1"
+  Scenario: As user, I can execute command on non active running profile 'minishift' from active profile 'foo'
```

Comment:
"As user, I can execute command on non active running profile 'minishift' from active profile 'foo'" -> "As user, I can execute a command against a non active profile"

Reply:
Short and precise